In [1]:
import pandas as pd
import numpy as np
import os
import shutil
from collections import Counter
from glob import glob

In [35]:
inpath = '/media/sc-mlsg/Brain2020/Registry_nii_sorted/'

In [36]:
seq_lst = []
for (path, dirs, files) in os.walk(inpath):
    if files:
        pt = path.split(os.path.sep)[-1]
        pt_seq_lst = [pt] + ['_'.join(file.split('.')[0].split("_")[1:]) for file in sorted(files)]
        seq_lst.append(pt_seq_lst)

In [37]:
seq = pd.DataFrame(seq_lst)
seq.to_csv('/media/sc-mlsg/Brain2020/Registry_nii/seq.csv')

In [24]:
# get all sequnece name
seq_lst = []
for (path, dirs, files) in os.walk(inpath):
    if files:
        pt_seq_lst = ['_'.join(file.split('.')[0].split("_")[1:]) for file in sorted(files)]
        seq_lst += pt_seq_lst

In [27]:
list(set(seq_lst))

['',
 '3D_Brain_View_FLAIR AX GD',
 'nNIPWI_Negative Integral',
 'nNI_negative integral',
 'nTTPPWI_Time To Peak',
 'V3D T1 BB iMSDE COR F GD',
 'T1 POST 3D COR F',
 '3D T1 TFE SAG PRE',
 't2_space_dark-fluid_sag_gd_MPR_Tra',
 'T2 FLAIR AX(11000_2800_110)',
 't1_dante_space_sag_bb_iso_gd',
 'T1 SAG',
 '3D T1 BB iMSDE COR F GD',
 '3D_Brain_View_FLAIR AX',
 't1_space_sag_BB_iso_MPR_Tra',
 't1_3D COR GD FS 1mm G2',
 't2_space_dark-fluid_sag_gd NEW',
 't1_space_sag_BB_iso_MPR_Cor',
 'sDWI b0',
 'DWI NEW THINCUT',
 'DWI 3mm',
 't1_space_3D_ BB_sag gd',
 'nINDEXnINDEXPWI_GRAY',
 '3D T1 TFE COR',
 'sDWI-B0',
 'T1 PRE 3D COR',
 '3D T1 TFE SAG F GD',
 'T1 PSOT 3D SAG F',
 'T2 AX GD',
 't2_tse_ax',
 '3D T1 TFE SAG',
 '3D T1 TFE COR F GD',
 'WIP SOURCE - 3D pCASL 192 ETL23 NEW',
 'T1 Post 3D AX F',
 '3D pCASL 192 ETL23 NEW',
 'T1 Post 3D COR F',
 '3D_Brain View FLAIR COR GD',
 't2_tse_fl2d_hemo',
 '3D T1 BB MSDE AX F GD',
 't2_space_dark-fluid_ax_gd NEW',
 'SAG T1',
 'T1 Post 3D SAG F',
 '3D T2 C

In [30]:
lst = Counter(seq_lst)
lst.most_common()

[('T2 AX', 234),
 ('T1 AX', 233),
 ('T2 FFE', 233),
 ('T1 SAG', 232),
 ('3D T1 TFE SAG F GD', 231),
 ('3D T1 TFE AX F GD', 221),
 ('3D T1 TFE COR F GD', 220),
 ('3D T1 BB iMSDE SAG F GD', 182),
 ('3D_Brain_View_FLAIR AX GD', 169),
 ('3D_Brain_View_FLAIR COR GD', 168),
 ('3D T1 BB MSDE AX F GD', 148),
 ('3D Brain View FLAIR SAG GD', 147),
 ('3D T1 BB MSDE COR F GD', 135),
 ('3D T1 BB iMSDE COR F GD', 86),
 ('T2 FLAIR AX(11000_2800_110)', 80),
 ('T2 FLAIR', 79),
 ('3D T1 BB iMSDE AX F GD', 74),
 ('3D_Brain View FLAIR SAG GD', 64),
 ('3D Brain View FLAIR COR GD', 52),
 ('3D Brain View FLAIR AX GD', 50),
 ('3D T1 BB iMSDE SAG F GD(compo)', 48),
 ('T2 FLAIR NEW', 46),
 ('SURVEY', 37),
 ('T2 FLAIR AX', 32),
 ('3D Brain FLAIR SAG GD', 22),
 ('DWI NEW', 21),
 ('DWI', 18),
 ('dADC', 16),
 ('V3D Brain View FLAIR AX GD', 13),
 ('V3D Brain View FLAIR COR GD', 13),
 ('V3D T1 BB iMSDE AX F GD', 13),
 ('V3D T1 BB iMSDE COR F GD', 13),
 ('V3D T1 TFE AX F GD', 13),
 ('V3D T1 TFE COR F GD', 13),
 ('sDWI

In [35]:
all_seq = pd.DataFrame.from_dict(lst, orient='index')
all_seq.to_csv('/media/sc-mlsg/Brain2020/Registry_nii/all_seq.csv')

T2 (total 242) ok
 ('T2 AX', 234),
 ('AX T2', 2),
 ('T2 AX GD', 1)
 ('3D T2 AX', 1),
 ('t2_tse_ax', 4),

 
T1 (total 243) ok
 ('T1 AX', 233),
 ('AX T1', 3),
 ('t1_tse_ax', 4),
 ('T1 PRE 3D AX', 1),
 ('3D T1 TFE AX', 2)

 
FLAIR (total 243) ok
 ('T2 FLAIR AX(11000_2800_110)', 80),
 ('T2 FLAIR', 79),
 ('T2 FLAIR NEW', 44),
 ('T2 FLAIR AX', 32),
 ('t2_tse_FLAIR_ax', 4),
 ('AX FLAIR', 4),

3D FLAIR (total 232) pass
 ('3D_Brain_View_FLAIR AX GD', 169),
 ('3D Brain View FLAIR AX GD', 50),
 ('V3D Brain View FLAIR AX GD', 13),
 ('3D_Brain_View_FLAIR AX', 1),

 
BB (total 235)
 ('3D T1 BB MSDE AX F GD', 148),
 ('3D T1 BB iMSDE AX F GD', 74),
 ('V3D T1 BB iMSDE AX F GD', 13),
 ('t1-space_3D_BB_ax gd', 1),
 ('t1_dante_space_sag_bb_iso_gd_MPR_Tra', 1)

WB (total 243) ok
 ('3D T1 TFE AX F GD', 221),
 ('V3D T1 TFE AX F GD', 13),
 ('3D T1 TFE AX GD', 2),
 ('T1 Post 3D AX F', 3),
 ('T1 POST 3D AX F', 1),
 ('t1_3D AX GD FS 1mm G2', 1)
 ('t1_3D SAG GD FS_MPR_Tra', 3)

In [102]:
inpath = '/media/sc-mlsg/Brain2020/Registry_nii_sorted/'

In [71]:
seq_lst = []
for (path, dirs, files) in os.walk(inpath):
    if files:
        pt = path.split(os.path.sep)[-1]
        pt_seq_lst = [pt] + ['_'.join(file.split('.')[0].split("_")[1:]) for file in sorted(files)]
        seq_lst.append(pt_seq_lst)

In [72]:
seq = pd.DataFrame(seq_lst)
seq.to_csv('/media/sc-mlsg/Brain2020/Registry_nii_sorted/seq.csv')

In [73]:
seq_lst

[['', ''],
 ['10304501', '3D_FLAIR', 'BB', 'FLAIR', 'T1', 'T2', 'WB'],
 ['10217353', '3D_FLAIR', 'BB', 'FLAIR', 'T1', 'T2', 'WB'],
 ['8805090', '3D_FLAIR', 'BB', 'FLAIR', 'T1', 'T2', 'WB'],
 ['8804808', '3D_FLAIR', 'BB', 'FLAIR', 'T1', 'T2', 'WB'],
 ['7341187', '3D_FLAIR', 'BB', 'FLAIR', 'T1', 'T2', 'WB'],
 ['5290867', '3D_FLAIR', 'BB', 'FLAIR', 'T1', 'T2', 'WB'],
 ['10350232', '3D_FLAIR', 'BB', 'FLAIR', 'T1', 'T2', 'WB'],
 ['9227858', 'FLAIR', 'T1', 'T2'],
 ['2161791', '3D_FLAIR', 'BB', 'FLAIR', 'T1', 'T2', 'WB'],
 ['10046518', '3D_FLAIR', 'BB', 'FLAIR', 'T1', 'T2', 'WB'],
 ['10136010', '3D_FLAIR', 'BB', 'FLAIR', 'T1', 'T2', 'WB'],
 ['5896524', '3D_FLAIR', 'BB', 'FLAIR', 'T1', 'T2', 'WB'],
 ['8670197', '3D_FLAIR', 'BB', 'FLAIR', 'T1', 'T2', 'WB'],
 ['10316513', '3D_FLAIR', 'BB', 'FLAIR', 'T1', 'T2', 'WB'],
 ['8775761', '3D_FLAIR', 'BB', 'FLAIR', 'T1', 'T2', 'WB'],
 ['10068329', '3D_FLAIR', 'BB', 'FLAIR', 'T1', 'T2', 'WB'],
 ['3388539', '3D_FLAIR', 'BB', 'FLAIR', 'T1', 'T2', 'WB'],
 ['

In [77]:
for i in seq_lst:
    print(i)

['', '']
['10304501', '3D_FLAIR', 'BB', 'FLAIR', 'T1', 'T2', 'WB']
['10217353', '3D_FLAIR', 'BB', 'FLAIR', 'T1', 'T2', 'WB']
['8805090', '3D_FLAIR', 'BB', 'FLAIR', 'T1', 'T2', 'WB']
['8804808', '3D_FLAIR', 'BB', 'FLAIR', 'T1', 'T2', 'WB']
['7341187', '3D_FLAIR', 'BB', 'FLAIR', 'T1', 'T2', 'WB']
['5290867', '3D_FLAIR', 'BB', 'FLAIR', 'T1', 'T2', 'WB']
['10350232', '3D_FLAIR', 'BB', 'FLAIR', 'T1', 'T2', 'WB']
['9227858', 'FLAIR', 'T1', 'T2']
['2161791', '3D_FLAIR', 'BB', 'FLAIR', 'T1', 'T2', 'WB']
['10046518', '3D_FLAIR', 'BB', 'FLAIR', 'T1', 'T2', 'WB']
['10136010', '3D_FLAIR', 'BB', 'FLAIR', 'T1', 'T2', 'WB']
['5896524', '3D_FLAIR', 'BB', 'FLAIR', 'T1', 'T2', 'WB']
['8670197', '3D_FLAIR', 'BB', 'FLAIR', 'T1', 'T2', 'WB']
['10316513', '3D_FLAIR', 'BB', 'FLAIR', 'T1', 'T2', 'WB']
['8775761', '3D_FLAIR', 'BB', 'FLAIR', 'T1', 'T2', 'WB']
['10068329', '3D_FLAIR', 'BB', 'FLAIR', 'T1', 'T2', 'WB']
['3388539', '3D_FLAIR', 'BB', 'FLAIR', 'T1', 'T2', 'WB']
['6125837', '3D_FLAIR', 'BB', 'FLAIR', 

T2(1): 10299525 
T1(2): 9115758 10258565
3D_FLAIR(11): 9227858 10271376 3487606 10308977 10299525 10052983 9115758 8840566 8767856 10258565 10251700

BB(9): 9227858 10271376 3487606 10299525 10052983 9115758 8767856 10258565 10251700
WB(4): 9227858 3487606 10052983 8840566

In [110]:
# find pt that does not exists in pt list
csvpath = '/home/sc-mlsg/Downloads/202001-KHS_mets.xlsx'
pt_list= pd.read_excel(csvpath)['등록번호'].to_list()

In [106]:
pat = []
for (path, dirs, files) in os.walk(inpath):
    if files:
        pt = path.split(os.path.sep)[-1]
        pat.append(pt)

In [116]:
total = 0
for pt in pt_list:
    total += 1
    if str(pt) not in pat:
        print(pt)

In [113]:
len(pat), len(pt_list)

(243, 246)

In [117]:
total

246

In [119]:
Counter(pt_list).most_common

<bound method Counter.most_common of Counter({5347425: 2, 10220640: 2, 10242913: 2, 10213385: 1, 9668839: 1, 3788918: 1, 5174594: 1, 7528825: 1, 10306380: 1, 10322968: 1, 7244642: 1, 2111194: 1, 10234091: 1, 10243245: 1, 10270019: 1, 9153397: 1, 10194609: 1, 8711005: 1, 10101924: 1, 10272845: 1, 5041222: 1, 3942957: 1, 5253497: 1, 10226877: 1, 10304501: 1, 8352615: 1, 8594953: 1, 10105301: 1, 8735084: 1, 8382957: 1, 10254064: 1, 8530251: 1, 8109880: 1, 10163496: 1, 7880821: 1, 10088298: 1, 8287959: 1, 4325578: 1, 2217165: 1, 4175603: 1, 8722052: 1, 10012333: 1, 10203648: 1, 7701021: 1, 5675407: 1, 10102852: 1, 8805090: 1, 8804808: 1, 10253622: 1, 8346378: 1, 10052983: 1, 5426030: 1, 4399069: 1, 10096058: 1, 10321208: 1, 2161791: 1, 10001984: 1, 8281761: 1, 10192623: 1, 8807758: 1, 8785610: 1, 8025354: 1, 4227645: 1, 10248520: 1, 10273560: 1, 10106403: 1, 10189417: 1, 7457329: 1, 8339811: 1, 7968246: 1, 8809163: 1, 7870544: 1, 10206232: 1, 8817504: 1, 10238419: 1, 10221986: 1, 10286743:

In [41]:
import pydicom
dcm_paths = glob('/media/sc-mlsg/Brain2020/Registry_dcm/*102429*/**/')
for dcm_path in dcm_paths:
    dcmData = pydicom.dcmread(os.path.join(dcm_path, os.listdir(dcm_path)[0]))
    study_date = dcmData[0x00080020].value
    print(study_date, dcm_path)

20200810 /media/sc-mlsg/Brain2020/Registry_dcm/10242913/V3D T1 TFE AX F GD/
20200810 /media/sc-mlsg/Brain2020/Registry_dcm/10242913/3D Brain View FLAIR SAG GD/
20200810 /media/sc-mlsg/Brain2020/Registry_dcm/10242913/V3D T1 TFE COR F GD/
20200810 /media/sc-mlsg/Brain2020/Registry_dcm/10242913/3D T1 BB iMSDE SAG F GD/
20200810 /media/sc-mlsg/Brain2020/Registry_dcm/10242913/T2 FFE/
20200810 /media/sc-mlsg/Brain2020/Registry_dcm/10242913/V3D T1 BB iMSDE AX F GD/
20200810 /media/sc-mlsg/Brain2020/Registry_dcm/10242913/V3D T1 BB iMSDE COR F GD/
20200810 /media/sc-mlsg/Brain2020/Registry_dcm/10242913/3D T1 TFE SAG F GD/
20200810 /media/sc-mlsg/Brain2020/Registry_dcm/10242913/T2 AX/
20200810 /media/sc-mlsg/Brain2020/Registry_dcm/10242913/V3D Brain View FLAIR COR GD/
20200810 /media/sc-mlsg/Brain2020/Registry_dcm/10242913/V3D Brain View FLAIR AX GD/
20200810 /media/sc-mlsg/Brain2020/Registry_dcm/10242913/T1 AX/
20200810 /media/sc-mlsg/Brain2020/Registry_dcm/10242913/T2 FLAIR/
20200810 /media/sc

In [26]:
dcm_paths

['/media/sc-mlsg/Brain2020/Registry_dcm/10242913/V3D T1 TFE AX F GD/',
 '/media/sc-mlsg/Brain2020/Registry_dcm/10242913/3D Brain View FLAIR SAG GD/',
 '/media/sc-mlsg/Brain2020/Registry_dcm/10242913/V3D T1 TFE COR F GD/',
 '/media/sc-mlsg/Brain2020/Registry_dcm/10242913/3D T1 BB iMSDE SAG F GD/',
 '/media/sc-mlsg/Brain2020/Registry_dcm/10242913/T2 FFE/',
 '/media/sc-mlsg/Brain2020/Registry_dcm/10242913/V3D T1 BB iMSDE AX F GD/',
 '/media/sc-mlsg/Brain2020/Registry_dcm/10242913/V3D T1 BB iMSDE COR F GD/',
 '/media/sc-mlsg/Brain2020/Registry_dcm/10242913/3D T1 TFE SAG F GD/',
 '/media/sc-mlsg/Brain2020/Registry_dcm/10242913/T2 AX/',
 '/media/sc-mlsg/Brain2020/Registry_dcm/10242913/V3D Brain View FLAIR COR GD/',
 '/media/sc-mlsg/Brain2020/Registry_dcm/10242913/V3D Brain View FLAIR AX GD/',
 '/media/sc-mlsg/Brain2020/Registry_dcm/10242913/T1 AX/',
 '/media/sc-mlsg/Brain2020/Registry_dcm/10242913/T2 FLAIR/',
 '/media/sc-mlsg/Brain2020/Registry_dcm/10242913/T1 SAG/',
 '/media/sc-mlsg/Brain20